feat(helm): support dual-stack networking for load balancers

bergemalm opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://github.com/googleforgames/agones/blob/main/CONTRIBUTING.md and developer guide https://github.com/googleforgames/agones/blob/main/build/README.md
2. Please label this pull request according to what type of issue you are addressing.
3. Ensure you have added or ran the appropriate tests for your PR: https://github.com/googleforgames/agones/blob/main/build/README.md#testing-and-building
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

This PR adds support for setting up dual-stack load balancers (IPv4 and IPv6).

gongmax commented:
/gcbrun

gongmax commented on install/helm/agones/values.yaml:
```diff
@@ -186,6 +186,10 @@ agones:
       loadBalancerIP: ""
       loadBalancerSourceRanges: []
       annotations: {}
+      # ipFamilyPolicy: RequireDualStack
+      # ipFamilies:
+      # - IPv4
+      # - IPv6
```
Instead of commenting out these settings, can we set them to empty values?

bergemalm replied:
What about setting them to what the default values?

```yaml
  ipFamilyPolicy: SingleStack
  ipFamilies:
  - IPv4
```

gongmax commented on install/helm/agones/values.yaml:
```diff
@@ -195,6 +199,10 @@ agones:
       loadBalancerIP: ""
       loadBalancerSourceRanges: []
       annotations: {}
+      # ipFamilyPolicy: RequireDualStack
+      # ipFamilies:
+      # - IPv4
+      # - IPv6
```
Same as above

gongmax commented on install/helm/agones/values.yaml:
```diff
@@ -238,6 +246,10 @@ agones:
       loadBalancerIP: ""
       loadBalancerSourceRanges: []
       annotations: {}
+      # ipFamilyPolicy: RequireDualStack
+      # ipFamilies:
+      # - IPv4
+      # - IPv6
```
Same as above

gongmax reviewed (commented):
Please also add descriptions in the relevant tables in https://github.com/googleforgames/agones/blob/main/site/content/en/docs/Installation/Install%20Agones/helm.md?plain=1

bergemalm commented:
PR updated. Descriptions added and empty values set. @gongmax

gongmax commented:
/gcbrun

gongmax commented:
/gcbrun

gongmax commented:
Sorry for the back and forth. The description needs to be in a `{{% feature publishVersion="1.46.0" %}}...{{% /feature %}}` block so it's only visible in next release. Similar as https://github.com/googleforgames/agones/commit/dc6d4a02a819820baa359542334785ac12daa8ac#diff-481b1839587cfc3a359e5e82a78b375445ebe75fe39d5c070150bc4f3a2f51adR195. Otherwise, LGTM

bergemalm commented:
> Sorry for the back and forth. We need to add expiry / publish versions on the documentation so it's only visible in next release. Similar as [dc6d4a0#diff-481b1839587cfc3a359e5e82a78b375445ebe75fe39d5c070150bc4f3a2f51adR195](https://github.com/googleforgames/agones/commit/dc6d4a02a819820baa359542334785ac12daa8ac#diff-481b1839587cfc3a359e5e82a78b375445ebe75fe39d5c070150bc4f3a2f51adR195). Otherwise, LGTM

No worries! Sorry for the late response (vacation times).
Update pushed. Let me know if there is anything more.

0xaravindh commented:
gcbrun

0xaravindh commented:
/gcbrun

bergemalm commented:
Hello! I've rebased on latest main and resolved conflicts. I updated the expiry/publish features too. Please have another look and let me know if anything more is required to have this released? Thanks @gongmax @0xaravindh
